cmake errors at -B containing @

Hello

When I try to issue a build directory that contains an @ sign, cmake complains; this is quite a problem because autobuilders such as make will issue @ as a workspace path for when running multiple jobs in parallel -- definitely breaks compatibility with previous behaviour.

```
$ mkdir build@4
$ cmake -DZEPHYR_GCC_VARIANT=zephyr -DBOARD=arduino_101 -H$ZEPHYR_BASE/samples/net/http_server/ -B$PWD/build@4/
...
-- Performing Test check_Wl___print_memory_usage
-- Performing Test check_Wl___print_memory_usage - Success
CMake Error at /home/inaky/z/kernel.git/cmake/extensions.cmake:310 (add_custom_target):
  The target name
  "_home_inaky_z_build@4_zephyr_include_generated_echo-apps-cert_der_inc" is
  reserved or not valid for certain CMake features, such as generator
  expressions, and may result in undefined behavior.
Call Stack (most recent call first):
  CMakeLists.txt:16 (generate_inc_file_for_target)


CMake Error at /home/inaky/z/kernel.git/cmake/extensions.cmake:310 (add_custom_target):
  The target name
  "_home_inaky_z_build@4_zephyr_include_generated_echo-apps-key_der_inc" is
  reserved or not valid for certain CMake features, such as generator
  expressions, and may result in undefined behavior.
Call Stack (most recent call first):
  CMakeLists.txt:16 (generate_inc_file_for_target)


CMake Error at /home/inaky/z/kernel.git/cmake/extensions.cmake:310 (add_custom_target):
  The target name
  "_home_inaky_z_build@4_zephyr_include_generated_index_html_inc" is reserved
  or not valid for certain CMake features, such as generator expressions, and
  may result in undefined behavior.
Call Stack (most recent call first):
  CMakeLists.txt:16 (generate_inc_file_for_target)


-- Configuring incomplete, errors occurred!
See also "/home/inaky/z/build@4/CMakeFiles/CMakeOutput.log".
See also "/home/inaky/z/build@4/CMakeFiles/CMakeError.log".

```

--- a/cmake/extensions.cmake
+++ b/cmake/extensions.cmake
@@ -306,6 +306,7 @@ function(generate_inc_file_for_target
 
   string(REPLACE "/" "_" generated_target_name ${generated_target_name})
   string(REPLACE "." "_" generated_target_name ${generated_target_name})
+  string(REPLACE "@" "_" generated_target_name ${generated_target_name})
 
   add_custom_target(${generated_target_name} DEPENDS ${generated_file})
   add_dependencies(${target} ${generated_target_name})

